lxd/instance/drivers/driver_qemu.go

```diff
@@ -3181,10 +3182,70 @@ func (d *qemu) addDriveConfig(bootIndexes map[string]int, driveConf deviceConfig
 		"read-only": false,
 	}
 
+	var secret string
+
 	// If driver is "file", QEMU requires the file to be a regular file.
 	// However, if the file is a character or block device, driver needs to be set to "host_device".
 	if isBlockDev {
 		blockDev["driver"] = "host_device"
+	} else if isRBDImage {
+		blockDev["driver"] = "rbd"
+		// Expected format is rbd:poolname/volumename:<options>
+		// where options is key=value:...
+		fields := strings.SplitN(driveConf.DevPath, ":", 3)
+
+		if len(fields) != 3 {
+			return nil, fmt.Errorf("Unexpected devPath rbd format %q for drive %q", driveConf.DevPath, driveConf.DevName)
+		}
+
+		opts := fields[2]
+		fields = strings.Split(fields[1], "/")
```

Comment:
Should this be strings.SplitN(fields[1], "/", 2) to show we only expect to get 2 fields?

Reply:
Also need to check the length of fields before using it after split here.

Reply:
Just to be 100%, `:` is not allowed in both pool name and volume name right?

Reply:
> Just to be 100%, `:` is not allowed in both pool name and volume name right?

```
$ lxc storage create local:foo:bar dir
Storage pool foo:bar created

$ lxc storage volume create default foo:bar
Storage volume foo:bar created
```

I guess we should handle that somehow :)

Reply:
Or we can just disallow `:` in pool and volume names, and hope that nobody is currently doing this.

Reply:
Storage pools cannot include space or `/` characters.

Reply:
Seems better to not do that as its an arbitrary limitation (that is going away once we move to using multiple static types for communicating config between device and instance).